src/main/kotlin/org/rust/ide/utils/StructFieldsExpander.kt

```diff
@@ -0,0 +1,55 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.utils
+
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.impl.source.tree.LeafPsiElement
+import org.rust.lang.core.psi.RsPatBinding
+import org.rust.lang.core.psi.RsPatStruct
+import org.rust.lang.core.psi.RsPatTupleStruct
+import org.rust.lang.core.psi.RsPsiFactory
+import org.rust.lang.core.psi.ext.RsFieldsOwner
+import org.rust.lang.core.psi.ext.childrenOfType
+import org.rust.lang.core.psi.ext.descendantsOfType
+import org.rust.lang.core.psi.ext.fields
+import org.rust.lang.core.resolve.ref.deepResolve
+import org.rust.openapiext.buildAndRunTemplate
+import org.rust.openapiext.createSmartPointer
+
+fun expand(project: Project, editor: Editor, body: PsiElement, startElement: PsiElement) {
+    var anchor = startElement.descendantsOfType<LeafPsiElement>().first { it.text == "{" }
+    while (anchor.nextSibling.text != "}") {
+        anchor.nextSibling.delete()
```

Comment:
It's a bad idea to delete all fields and fill them again b/c there can be comments or specific formatting. Take a look at e.g. `AddStructFieldsFix` (`calculateMissingFields`/`fillStruct`)